app/models/host_status/execution_status.rb

```diff
@@ -0,0 +1,36 @@
+class HostStatus::ExecutionStatus < HostStatus::Status
+  OK = 0
+  ERROR = 1
+
+  def to_status(options = {})
+    task = ForemanTasks::Task::DynflowTask.for_action(Actions::RemoteExecution::RunHostJob).for_resource(host).order(:started_at).where(:state => 'stopped').last
+    if task.nil? || task.result == 'success'
+      OK
+    else
+      ERROR
+    end
+  end
+
+  def to_global(options = {})
+    if to_status(options) == ERROR
+      return HostStatus::Global::ERROR
+    else
+      return HostStatus::Global::OK
+    end
+  end
+
+  def self.status_name
+    N_('Execution')
+  end
+
+  def to_label(options = {})
+    case to_status(options)
+      when OK
+        N_('Last execution succeeded')
```

Comment:
Sorry one last thought, a host with no executions will have a status that says `Last execution succeeded`, would that be confusing?

Does it make sense to have a different status for that? Like "No executions" that's rated as a Global::OK?